processing/src/main/java/io/druid/segment/ColumnSelectorBitmapIndexSelector.java

```diff
@@ -115,7 +115,56 @@ public BitmapFactory getBitmapFactory()
   public BitmapIndex getBitmapIndex(String dimension)
   {
     final Column column = index.getColumn(dimension);
-    if (column != null && column.getCapabilities().hasBitmapIndexes()) {
+
+    if (column == null) {
+      return new BitmapIndex() {
+
+        @Override
+        public int getCardinality()
+        {
+          return 1;
+        }
+
+        @Override
+        public String getValue(int index)
+        {
+          return null;
+        }
+
+        @Override
+        public boolean hasNulls()
+        {
+          return true;
+        }
+
+        @Override
+        public BitmapFactory getBitmapFactory()
+        {
+          return bitmapFactory;
+        }
+
+        @Override
+        public int getIndex(String value)
+        {
+          // Return -2 for non-null values to match what the BitmapIndex implementation in BitmapIndexColumnPartSupplier
+          // would return for getIndex() when there is only one index, for the null value.
+          // i.e., return an 'insertion point' of 1 for non-null values (see BitmapIndex interface)
+          return Strings.isNullOrEmpty(value) ? 0 : -2;
+        }
+
+        @Override
+        public ImmutableBitmap getBitmap(int idx)
+        {
+          if (idx == 0) {
+            return bitmapFactory.complement(bitmapFactory.makeEmptyImmutableBitmap(), getNumRows());
+          } else {
+            return bitmapFactory.makeEmptyImmutableBitmap();
+          }
+        }
+      };
+    }
+
+    if (column.getCapabilities().hasBitmapIndexes()) {
```

Comment:
I think `else if` would be clearer.

Why are we returning this BitmapIndex impl for columns that don't exist? I know you told me but I forgot :). It should probably be a comment.